src/main/OP.ts

```diff
@@ -140,6 +145,12 @@ export class OP {
 async function parseAndResolveUri(encodedUri: string) {
   const requestPayload = AuthenticationRequest.parseURI(encodedUri);
   const jwt = requestPayload.request || (await (await fetch(requestPayload.request_uri)).text());
+
+  AuthenticationRequest.assertValidRequestObject(requestPayload);
+  AuthenticationRequest.assertValidRegistrationObject(
+    await AuthenticationRequest.getRegistrationObj(requestPayload.registration_uri, requestPayload.registration)
```

Comment:
Extract this into an assignment, as you now see what I warned about elsewhere. If this is registration by Ref, we are now hitting the same URI twice in this method (see below).
Actually the line below does the same as the getRegistrationObj method

Reply:
for this issue, I extracted a method hierarchy call for `getRegistrationObj` and changed it to call it once (per separate method call)